modules/packages/LinearAlgebra.chpl

```diff
@@ -1977,6 +2001,42 @@ module Sparse {
     return A;
   }
 
+  /* Adds a diagonal to a square sparse array by adding indices to
+      its sparse domain and sets the value of that diagonal to a 
+      constant. 
+    
+      ``k > 0``, represents an upper diagonal starting
+      from the ``k``th column, ``k == 0`` represents the main 
+      diagonal, ``k < 0`` represents a lower diagonal starting
+      from the ``-k``th row. ``k`` is 0-indexed.
+  */
+  proc setDiag (ref X: [?D] ?eltType, in k: int = 0, val: eltType = 0)
```

Comment:
minor: I don't think the `in` intent is necessary here, but it's also harmless to leave in.